### ISSUE TITLE
uh oh it broke

### ISSUE THREAD

**Issue by woodybury** [jamesloper/poon-router#1]
I believe this logic is wrong:

```
	if (currentScreen.route === route && currentScreen.pathname === location.pathname) { // try to stay on existing screen first
		console.log('update route:', location.pathname);
		currentScreen.setRoute(route, location);	
```

having issues w/ setQueryParams on homepage instead of staying on existing screen, its duplicating the "/" route. I looked into it and `currentScreen.pathname` is undefined, `currentScreen.pathNameStore.state` works:

e.g.
```
	if (currentScreen.route === route && currentScreen.pathNameStore.state === location.pathname) { // try to stay on existing screen first
```

**Comment by woodybury** [comment]
```
diff --git a/node_modules/poon-router/index.js b/node_modules/poon-router/index.js
index 051a5f7..f87d9d6 100644
--- a/node_modules/poon-router/index.js
+++ b/node_modules/poon-router/index.js
@@ -32,7 +32,7 @@ window.onpopstate = e => {
 	const route = routes.find(route => route.test(location.pathname));
 	const currentScreen = getTopScreen();
 
-	if (currentScreen.route === route && currentScreen.pathname === location.pathname) { // try to stay on existing screen first
+	if (currentScreen.route === route && currentScreen.pathNameStore.state === location.pathname) { // try to stay on existing screen first
 		console.log('update route:', location.pathname);
 		currentScreen.setRoute(route, location);
 	} else { // change state index
```

**Comment by woodybury** [comment]
PR https://github.com/jamesloper/poon-router/pull/2

**Comment by lemontron** [comment]
merged and published as `poon-router@1.1.3`
